### ISSUE TITLE
Store PA Profile with Filament Profile

### ISSUE THREAD

**Issue by greghesp** [bambulab/BambuStudio#2548]
The new calibration options in Studio are great, but their setup and configuration make very little sense and are pretty confusing.

In the Prepare tab, I have Filament settings.  These settings let you control everything related to the Filament, essentially giving you a custom profile, per filament.  This is what I use to set my filaments
![image](https://github.com/bambulab/BambuStudio/assets/768908/0366a168-0c29-462a-bbb2-c3130e9b77cb)

Now head over to the Device tab, and here we have the filaments that are in our AMS, which *should* be synced to the filaments above:
![image](https://github.com/bambulab/BambuStudio/assets/768908/70dda217-bbd1-4b24-8635-19133044709a)

If we click the Edit button here, things now get confusing.  My filament is now "Generic Filament" and my custom filament profile doesn't exist.  In addition, there is now a PA  Profile which applies the K value, if I manually apply it:

![image](https://github.com/bambulab/BambuStudio/assets/768908/9e66193d-6c43-462d-b378-6d13198fae38)
![image](https://github.com/bambulab/BambuStudio/assets/768908/8746df8a-cc14-44df-b58d-aef2a631ce5d)

Note:  The PA Profile has the same name as the Filament Profile I made.  They are not kept in sync, and have no relation to each other

None of this is kept in sync, and it seems that the PA profile is also lost, which needs reassigning all the time.

**Suggestion**

Would it not make sense, to just have 1 way to control this which is Filament Profiles as seen in the first screenshot?  If you go through manual calibration, you can save that PA profile, against the Filament Profile.  No separate setting, no losing values.  Just 1 simple way to manage all your filaments.

**Comment by simonvizzini** [comment]
That's actually how OrcaSlicer handles the PA setting, within the Filament profile. Would love if BambuStudio would do the same.

**Comment by threadstone83** [comment]
+1

**Comment by Danitegue** [comment]
+1

**Comment by gpeden** [comment]
this seems to be broken or inconsistent for me. Sometimes the PA Profile is set and in the drop down and the k value is set, then later it is gone -the k value and there is no PA profile in the drop down.  Am I doing something wrong?  Its frustring to go through the work of calibrating filament only to have it not work.  Its probably my fault.

**Comment by Danitegue** [comment]
Same happens to me. Cannot retain calibrations after restarting the machine. The drop-down to select the K and PA values are often empty, and sometimes we cannot choose any profile or it is not showing all possible profiles of all the previous calibrations I have done. It is really annoying to start calibrating from zero again, it is a waste of time and filament. We really need that bambulab create a clear unique guide in order to explain how the users should do to have a database of properly calibrated filaments. In this guide they could explain also how to deal with the limitations of the machine. (I.e. if we cannot store everything into machine memory, then explain when is the machine using the last config and when not). Also they could be more transparent: show somewhere in the printer screen, what are the actual K and PA values being used, so that we can identify since the start of the print if the printer is using the correct one. This would allow to identify the problem and abort printing before the print starts, instead of having a wrongly printed piece with ugly last layer aspect, after a few lost hours...

**Comment by zhimin-zeng-bambulab** [comment]
@gpeden @Danitegue  If you encounter the problem of K value disappearing next time, could you share the log files?

**Comment by SaltWei** [comment]
> 99% of the time, what you're selecting in the slicer is what you're going to print with. If people want to slice and prep with something not in the AMS or for another printer, let them override it.

Because the PA value of the same material can vary on different machines, and it can also change over time as the nozzle is used, storing the PA value in a preset can disrupt the ability to directly print a 3MF file with a single click.

**Comment by Danitegue** [comment]
> > 99% of the time, what you're selecting in the slicer is what you're going to print with. If people want to slice and prep with something not in the AMS or for another printer, let them override it.
> 
> Because the PA value of the same material can vary on different machines, and it can also change over time as the nozzle is used. Storing the PA value in a preset can disrupt the ability to directly print a 3MF file with a single click on app

Yes the PA calibrated for one specific filament can vary lightly along the time, and also for different slots of the AMS. But it varies much more if the machine suddenly uses a PA calculated for another filament, because the machine resets the PA values after a power reset... or is simply using the last one which might not be appropriate for a new filament. I would say one of the main problems is that we have no idea about which PA is really using the printer. 

One idea would be to add four PA presets in the filament profile: one for each AMS slot. Then the printer would use the appropriate one, depending on which AMS slot is mounted the filament. Then the PA calibration would update the appropriate PA preset in the filament profile.

With respect the ability to directly print from an 3MF file, I would say nobody using a professional bambu is using the direct printing from a 3MF file, there are always minor customizations needed, like changing the filament of the 3MF file to the filament I have available, and calibrated.

**Comment by redcaptm** [comment]
+1
I sepnt so much time with calibration, now I have a custom profile. I also manually added the filament and needed to copy the custom profile!? now i have 2 of them and when I setup the AMS, i could not choose any of these two???
For what i invest so much effort?

**Comment by mohoo** [comment]
+1
